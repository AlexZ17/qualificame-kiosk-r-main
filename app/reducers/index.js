import Auth, * as fromAuth from 'qualificame-kiosk/app/reducers/AuthReducer.js';
import Kiosk, * as fromKiosk from 'qualificame-kiosk/app/reducers/KioskReducer.js';
import { combineReducers } from 'redux';
const rootReducer = combineReducers({
    Auth,
    Kiosk
});

export default rootReducer;

//Auth
export const getIsLoggedIn = state =>
	fromAuth.getIsLoggedIn(state.Auth);

export const getIsLoading = state =>
	fromAuth.getIsLoading(state.Auth);

export const getAuthData = state =>
	fromAuth.getAuthData(state.Auth);

export const getUserData = state =>
	fromAuth.getUserData(state.Auth);

//Kiosk
export const getKiosk = state =>
	fromKiosk.getKiosk(state.Kiosk);

export const getQuestions = state =>
	fromKiosk.getQuestions(state.Kiosk);