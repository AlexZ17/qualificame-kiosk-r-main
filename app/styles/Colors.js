const Colors = {
	black: '#000000',
	blue: '#146DCA',
	dark: '#364350',
	gray: '#ccc',
	green: '#14A286',
	orange: '#EC480C',
	pink: '#F41C52',
	red: '#EB3F3F',
	white: '#FFFFFF',
	purple: '#A1258A',
	
	lightBlue: '#DCE9F7',
	lightGreen: '#DAF0E2',
	lightOrange: '#FCE4DA',
	lightPink: '#FDDDE5',
	
	solidGreen: '#27977F',

	
	transparent: 'rgba(0,0,0,0)',
	disabled: '#E0E0E0',
};

export default Colors;
