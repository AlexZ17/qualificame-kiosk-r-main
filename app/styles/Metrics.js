import { Dimensions, Platform, StatusBar } from 'react-native';

const { width, height } = Dimensions.get('window');

// Used via Metrics.baseMargin
const Metrics = {
	tinyMargin: 4, 
	smallMargin: 8,
	baseMargin: 16,
	doubleBaseMargin: 32,
	
	screenWidth: width < height ? width : height,
	screenHeight: width < height ? height : width,
	navBarHeight: (Platform.OS === 'ios') ? 64 : 79,
	statusBarHeight: StatusBar.currentHeight,
	
	buttonRadius: 10,
	
	icons: {
		small: 16,
		regular: 24,
		large: 48,
	}
};

export default Metrics;
