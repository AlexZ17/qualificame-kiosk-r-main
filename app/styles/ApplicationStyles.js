import Fonts from './Fonts'
import Colors from './Colors'
import Metrics from './Metrics'
import { StatusBar, Platform } from 'react-native';

// This file is for a reusable grouping of Theme items.
// Similar to an XML fragment layout in Android

const ApplicationStyles = {
	screen: {
		imageFill: {
			position:'absolute', 
			top: 0, 
			bottom: 0, 
			right: 0, 
			left: 0, 
			width:null, 
			height:null		
		},
		fill: {
			width: '100%',
			height: '100%'
		},
		container: {
			flex: 1
		},
		card: {
			marginVertical: Metrics.smallMargin,
			padding: Metrics.smallMargin,
			backgroundColor: Colors.white,
			shadowColor: '#000',
			shadowRadius: 4,
			shadowOpacity: 0.1,
			shadowOffset: { width: 0, height: 3 },
			elevation: 1,
			borderRadius: 5,
		},
		absolute: {
			position: 'absolute'
		},
		setToBot: {
			bottom: 0,
			right: 0,
			left: 0
		},
		behind: {
			zIndex: -1
		},
		headerImage: {
			width: Metrics.screenWidth,
			height: (Metrics.screenWidth * .46 ) 
		},
		footerImage: {
			width: Metrics.screenHeight,
			height: (Metrics.screenWidth * .25 ) 
		},
		customSpacer: {
			marginTop: (Metrics.screenWidth * .46 ) - (Metrics.navBarHeight + 27),
		}
	},
	margins: {
		tinyVerticalMargin: {
			marginVertical: Metrics.tinyMargin
		},
		smallVerticalMargin: {
			marginVertical: Metrics.smallMargin
		},
		baseVerticalMargin: {
			marginVertical: Metrics.baseMargin
		},
		smallHorizontalMargin: {
			marginHorizontal: Metrics.smallMargin
		},
		baseHorizontalMargin: {
			marginHorizontal: Metrics.baseMargin
		},
		smallTopMargin: {
			marginTop: Metrics.smallMargin
		},	
		baseTopMargin: {
			marginTop: Metrics.baseMargin
		},	
		dobleBaseTopMargin: {
			marginTop: Metrics.doubleBaseMargin
		},
		tinyRightMargin: {
			marginRight: Metrics.tinyMargin
		},
		smallRightMargin: {
			marginRight: Metrics.smallMargin
		},
		baseRightMargin: {
			marginRight: Metrics.baseMargin
		},
		smallLeftMargin: {
			marginLeft: Metrics.smallMargin
		},
		baseLeftMargin: {
			marginLeft: Metrics.baseMargin
		},
		smallBottomMargin: {
			marginBottom: Metrics.smallMargin
		},
		baseBottomMargin: {
			marginBottom: Metrics.baseMargin
		},
		doubleBaseBottomMargin: {
			marginBottom: Metrics.doubleBaseMargin
		},
		statusBarSpace: {
			paddingTop: (Platform.OS != 'ios') ? StatusBar.currentHeight : 0
		},
		navBarHeight: {
			height: Metrics.navBarHeight,
		},
		basePaddingAround: {
			padding: Metrics.baseMargin
		},
		tinyVerticalPadding: {
			paddingVertical: Metrics.tinyMargin
		},		
		smallVerticalPadding: {
			paddingVertical: Metrics.smallMargin
		},
		baseVerticalPadding: {
			paddingVertical: Metrics.baseMargin
		},
		smallHorizontalPadding: {
			paddingHorizontal: Metrics.smallMargin
		},
		baseHorizontalPadding: {
			paddingHorizontal: Metrics.baseMargin
		},
		footerMargin: {
			marginBottom: (Metrics.screenWidth * .25 )
		},
		smallBottomPadding: {
			paddingBottom: Metrics.smallMargin
		},
		baseBottomPadding: {
			paddingBottom: Metrics.baseMargin
		},
		doubleBaseBottomPadding: {
			paddingBottom: Metrics.doubleBaseMargin
		}
	}, 
	flexBox: {
		column: {
			flexDirection: 'column'
		},
		row: {
			flexDirection: 'row'
		},
		flex1: {
			flex: 1
		},
		flex2: {
			flex:2
		},
		flex: {
			flex:3
		},
		justifyContentFlexStart: {
			justifyContent: 'flex-start'
		},
		justifyContentFlexEnd: {
			justifyContent: 'flex-end'
		},
		justifyContentCenter: {
			justifyContent: 'center'
		},
		justifyContentSpaceBetween: {
			justifyContent: 'space-between'
		},
		justifyContentSpaceAround:{
			justifyContent: 'space-around'
		},
		justifyContentSpaceEvenly:{
			justifyContent: 'space-evenly'
		},
		alignItemFlexStart: {
			alignItems: 'flex-start'
		},
		alignItemsFlexEnd: {
			alignItems: 'flex-end'
		},
		alignItemsCenter: {
			alignItems: 'center'
		},
		alignItemsStretch: {
			alignItems: 'stretch'
		},
		centerObjects: {
			justifyContent: 'center',
			alignItems: 'center'
		},
		alignSelfFlexStart: {
			alignSelf: 'flex-start'
		},
		alignSelfCenter: {
			alignSelf: 'center'
		},
		wrap: {
			flexWrap: 'wrap'
		},
		alignContentFlexStart: {
			alignContent: 'flex-start'
		},
		alignContentFlexEnd: {
			alignContent: 'flex-end'
		},
		alignContentCenter: {
			alignContent: 'center'
		},
		alignContentSpaceBetween: {
			alignContent: 'space-between'
		},
		alignContentSpaceAround: {
			alignContent: 'space-around'
		},
		fullWidth: {
			width: '100%'
		},
		fullHeigth: {
			height: '100%'
		},
		deviceHeight: {
			height: Metrics.screenHeight
		}
	},
	stackNavigatorOptions: {
		removeHeader: {
			headerShown: false
		},
		transparentHeader: {
			headerTransparent: true,
			headerStyle: {
				backgroundColor: Colors.transparent,
				elevation: 0, // remove shadow on Android
				shadowOpacity: 0, // remove shadow on iOS
				borderBottomWidth: 0,
			},
			gesturesEnabled: false,
			headerTintColor: Colors.white
		},
		bottomTab: {
			activeTintColor: Colors.blue,
			inactiveTintColor: Colors.light,
			style: {
				backgroundColor: Colors.white
			}
		}
	}
} 

export default ApplicationStyles
