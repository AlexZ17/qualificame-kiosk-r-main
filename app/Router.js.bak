// Use this file to create all app navigators.

import React, { Component } from 'react';
import { createAppContainer, createSwitchNavigator } from 'react-navigation';
import {createBottomTabNavigator} from 'react-navigation-tabs';
import {createStackNavigator} from 'react-navigation-stack';

import WelcomeScreen from 'qualificame-kiosk/app/screens/WelcomeScreen';
import ScannerScreen from 'qualificame-kiosk/app/screens/ScannerScreen';
import MainScreen from 'qualificame-kiosk/app/screens/MainScreen';
import DetailScreen from 'qualificame-kiosk/app/screens/DetailScreen';
import LogoutScreen from 'qualificame-kiosk/app/screens/LogoutScreen';
import CompletedScreen from 'qualificame-kiosk/app/screens/CompletedScreen';
import AuthLoadingScreen from 'qualificame-kiosk/app/screens/AuthLoadingScreen';
import TransitionConfiguration from 'qualificame-kiosk/app/utils/TransitionConfiguration';
import { ApplicationStyles } from 'qualificame-kiosk/app/styles';

const ModalNavigator = createStackNavigator({
	WelcomeScreen,
	ScannerScreen
}, {
	mode: 'modal',
	navigationOptions: {
		header: null,
	},
	transitionConfig: TransitionConfiguration
})

const AppNavigator = createStackNavigator({
	MainScreen,
	DetailScreen,
	CompletedScreen,
	LogoutScreen,
}, {
	initialRouteName: 'MainScreen',
	navigationOptions: {
		header: null
	},
	transitionConfig: TransitionConfiguration
});

const MainNavigation = createSwitchNavigator({
	AuthLoading: AuthLoadingScreen,
	App: AppNavigator,
	Auth: ModalNavigator
},{
	initialRouteName: 'AuthLoading',
});


export default createAppContainer(MainNavigation);
