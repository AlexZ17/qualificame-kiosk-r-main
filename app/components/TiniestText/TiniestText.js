import React, { PureComponent } from 'react';
import { Text, View } from 'react-native';
import PropTypes from 'prop-types';

import styles from './TiniestTextStyle';
import { Fonts, Colors } from 'qualificame-kiosk/app/styles';

export default class TiniestText extends PureComponent {
	render() {
		const textColor = { color: Colors[this.props.color] };
		const weight = { fontWeight: Fonts.weight[this.props.weight] };
		const align = { textAlign: this.props.align };
		
		return (
			<View>
				<Text style={[styles.tiniestText, textColor, weight, align]}>
					{this.props.text}
				</Text>
			</View>
		);
	}
}

TiniestText.propTypes = {
	text: PropTypes.string,
	color: PropTypes.string,
	weight: PropTypes.string,
	align: PropTypes.string,
}

TiniestText.defaultProps = {
	text: '',
	color: "black",
	weight: "regular",
	align: "left"
}