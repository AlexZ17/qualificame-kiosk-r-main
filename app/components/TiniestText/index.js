import TiniestText from './TiniestText';

export default TiniestText;
