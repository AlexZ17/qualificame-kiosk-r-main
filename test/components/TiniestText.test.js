import React from 'react';
import { shallow } from 'enzyme';

import TiniestText from 'qualificame-kiosk/app/components/TiniestText';
import { Text } from 'react-native';
import { Fonts, Colors } from 'qualificame-kiosk/app/styles';

function createTestProps (props) {
	return {
		// common props
		color: '',
		text: 'testing',
		weight: '',
		align: '',
		// allow to override common props
		...props,
	}
}

describe('rendering', () => {
	let wrapper;
	const createWrapper = props => shallow(<TiniestText {...props}/>);
	beforeEach(() => {
		const props = createTestProps()
		wrapper = createWrapper(props);
	});
	it('renders as expected', () => {
		expect(wrapper).toMatchSnapshot();
	});
	it('renders text correctly', () => {
		expect(wrapper.find(Text).children().text()).toEqual('testing');
	});
	it('renders text with correct font color', () => {
		expect(wrapper.find(Text).props().style.find(p => p.color).color).toEqual(Colors.black);
	});
	it('renders text with correct font weight', () => {
		expect(wrapper.find(Text).props().style.find(p => p.fontWeight).fontWeight).toEqual(Fonts.weight.regular);
	});
	it('renders text with correct font size', () => {
		expect(wrapper.find(Text).props().style.find(p => p.fontSize).fontSize).toEqual(Fonts.size.tiniest);
	});
	it('renders text with correct alignment', () => {
		expect(wrapper.find(Text).props().style.find(p => p.textAlign).textAlign).toEqual('left');
	});
});