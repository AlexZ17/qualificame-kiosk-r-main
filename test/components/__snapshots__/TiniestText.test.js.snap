// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering renders as expected 1`] = `
<Component>
  <Component
    style={
      Array [
        Object {
          "fontSize": 7,
        },
        Object {
          "color": "#000000",
        },
        Object {
          "fontWeight": "400",
        },
        Object {
          "textAlign": "left",
        },
      ]
    }
  >
    testing
  </Component>
</Component>
`;
